generators/generator-adaptive-bot/generators/app/index.js

```diff
@@ -71,24 +71,64 @@ module.exports = class extends Generator {
     _validateIncludeApplicationSettings(opts) {
         let result = true;
         if ('includeApplicationSettings' in opts &&
-            typeof(opts.includeApplicationSettings) === 'boolean') {
+            typeof opts.includeApplicationSettings === 'boolean') {
             result = opts.includeApplicationSettings;
         }
 
         return result;
     }
 
     _validatePackageReferences(packageReferences) {
-        let result = [];
+        let results = [];
         if (Array.isArray(packageReferences)) {
             packageReferences.forEach((reference) => {
-                if (typeof reference == 'object' && reference.name && reference.version) {
-                    result.push(reference);
+                if (typeof reference === 'object' &&
+                    'name' in reference &&
+                    reference.name &&
+                    typeof reference.name === 'string' &&
+                    'version' in reference &&
+                    reference.version &&
+                    typeof reference.version === 'string') {
+
+                    const result = {
+                        name: reference.name,
+                        version: reference.version
+                    };
+
+                    results.push(result);
                 }
             });
         }
 
-        return result;
+        return results;
+    }
+
+    _validatePluginDefinitions(pluginDefinitions) {
+        let results = [];
+        if (Array.isArray(pluginDefinitions)) {
+            pluginDefinitions.forEach((definition) => {
+                if (typeof definition === 'object' &&
+                    'name' in definition &&
+                    definition.name &&
+                    typeof definition.name === 'string') {
+
+                    const result = {
+                        name: definition.name
+                    };
+
+                    if ('settingsPrefix' in definition &&
+                        definition.settingsPrefix &&
+                        typeof definition.settingsPrefix === 'string') {
+
+                        result.settingsPrefix = definition.settingsPrefix;
+                    }
```

Comment:
Based on our chat yesterday, should we have an `else` statement here that sets result.settingsPrefix = definition.name?